craft_parts/utils/os_utils.py

```diff
@@ -172,13 +172,9 @@ def is_snap(*, application_name: Optional[str] = None) -> bool:
     if application_name:
         res = snap_name == application_name
     else:
-        res = snap_name is not None
+        res = snap_name != ""
```

Comment:
Generally i'd suggest using the default of None to distinguish if envvar is actually set to empty string.

```
$ FOO="" python -c 'import os; print(repr(os.environ.get("FOO")))'
''
$ python -c 'import os; print(os.environ.get("FOO"))'
None
```

But in this case it's non-consequential, so I'm ok as-is.

Reply:
I used the empty string to avoid an extra test before comparing to `application_name`.

Reply:
what if SNAP_NAME is set but does not match application name?

- Launch emacs/vscode snap
- start vterm
- env will have SNAP_NAME, but not the SNAP_NAME we want

we don't care if SNAP_NAME is set, only from the point of view of it matching the application name, I believe my simpler check does the trick.

Reply:
This is an interesting issue because craft-parts doesn't know the snap name of the application it's running under, so perhaps it should check if SNAP_NAME is set to something, regardless of the name.